lib/AST/StmtPrinter.cpp

```diff
@@ -1735,6 +1735,46 @@ void StmtPrinter::VisitAtomicExpr(AtomicExpr *Node) {
   OS << ")";
 }
 
+// Checked C extension
+
+void StmtPrinter::VisitCountBoundsExpr(CountBoundsExpr *Node) {
+  switch (Node->getKind()) {
+    case CountBoundsExpr::Kind::Byte_Count:
+      OS << "byte_count(";
+    case CountBoundsExpr::Kind::Count:
+      OS << "count(";
+    default:
+      llvm_unreachable("count bounds kind not covered by switch");
+      break;
+  }
+  PrintExpr(Node->getCountExpr());
+  OS << ")";
+}
+
+void StmtPrinter::VisitNullaryBoundsExpr(NullaryBoundsExpr *Node) {
+  switch (Node->getKind()) {
+  case NullaryBoundsExpr::Invalid: {
+    OS << "bounds(invalid)";
```

Comment:
Hmm--seems like no good answer here.  This is not a straightforward AST dumper--it's a pretty printer.  And for a previous project I used it to round-trip code from .C -> AST -> .C (back into parseable language text); anytime I saw it emit invalid C code I had to apply a workaround (and assumed it was a code defect).  And I see that there is a precedent of using llvm_unreachable in various places in this file, but I agree that might not be so great to invoke for someone who really does manually construct an AST with an invalid node like this.

Ultimately, I'll defer to your judgment on this.

---
In reply to: [69020519](https://github.com/Microsoft/checkedc-clang/pull/8#discussion_r69020519) [](ancestors = 69020519)